SofaKernel/framework/sofa/core/loader/BaseLoader.cpp

```diff
@@ -46,10 +46,14 @@ BaseLoader::~BaseLoader()
 void BaseLoader::parse(sofa::core::objectmodel::BaseObjectDescription *arg)
 {
     objectmodel::BaseObject::parse(arg);
+
+    bool success = false;
     if (canLoad())
-        load();
-    else
-        sout << "Doing nothing" << sendl;
+        success = load();
+    
+    // File not loaded, component is set to invalid
+    if (!success)
+        m_componentstate = sofa::core::objectmodel::ComponentState::Invalid;
```

Comment:
Interesting... in general in all my components I change the state only in init() because in my mind init() is the only function that allows a component to switch from an invalid/unknow state into a valid state. 

Classical pattern I use in init(): 
```cpp
    m_componentstate = ComponentState::Invalid;

    ... do all the checking....
    if(something_to_check_failure)
          return ;
    
   m_componentstate = Valid
```

So I never check that the state may be invalid before init. I don't know if this is a problem.

Reply:
The thing is several tests in the current Mesh api is done at "CanCreate" step.
But in fact if canCreate failed I assume the component won't be created and thus m_componentState will not be reachable.